lib/upload-s3.js

```diff
@@ -1,49 +1,57 @@
-const AWS = require('aws-sdk')
+const { Upload } = require('@aws-sdk/lib-storage')
+const { S3, S3Client } = require('@aws-sdk/client-s3')
 const Timer = require('./timer')
+const { PassThrough, pipeline } = require('stream')
 
-// configure AWS to log to stdout
-AWS.config.update({
-    logger: process.stdout
-})
-
-module.exports = function (stream, config, key) {
+module.exports = async (stream, { S3_BUCKET, S3_REGION, S3_STORAGE_CLASS, S3_URL }, Key) => {
     if (!stream || typeof stream.on !== 'function') {
         throw new Error('invalid stream provided')
     }
-    return new Promise((resolve, reject) => {
-        const timer = new Timer();
+
+    const passThrough = new PassThrough()
+    pipeline(stream, passThrough, () => {})
```

Comment:
Apparently the new sdk expects a Stream, so we create one here (more info: https://github.com/aws/aws-sdk-js-v3/issues/2522#issuecomment-939457957)